### PR TITLE
Add support for larger retention than prom1.x staleness delta

### PR DESCRIPTION
**Pull request by Thib17**
If set, MaxPointDelta is used to linear interpolate intermadiate points.

### CODE REVIEW

File: client/graphite/config/cli.go [hunk 1]
```diff
@@ -14,6 +14,10 @@ func AddCommandLine(app *kingpin.Application, cfg *Config) {
 		"The URL of the remote Graphite Web server to send samples to.").
 		StringVar(&cfg.Read.URL)
 
+	app.Flag("graphite.read.max-point-delta",
+		"If set, interval used to linear interpolate intermadiate points.").
```
Comment (iksaif) [inline client/graphite/config/cli.go]: linearly interpolate

Comment (iksaif) [reply client/graphite/config/cli.go]: intermediate

File: client/graphite/read.go [hunk 2]
```diff
@@ -192,12 +193,27 @@ func (c *Client) targetToTimeseries(ctx context.Context, target string, from str
...
 		}
-		for _, datapoint := range renderResponse.Datapoints {
-			timstampMs := datapoint.Timestamp * 1000
+		for j, datapoint := range renderResponse.Datapoints {
+			timestampMs := datapoint.Timestamp * 1000
 			if datapoint.Value == nil {
 				continue
 			}
-			ts.Samples = append(ts.Samples, &prompb.Sample{Value: *datapoint.Value, Timestamp: timstampMs})
+			ts.Samples = append(ts.Samples, &prompb.Sample{Value: *datapoint.Value, Timestamp: timestampMs})
```
Comment (iksaif) [inline client/graphite/read.go]: can you put that in another function to avoid going too much to the right ?

File: client/graphite/read.go [hunk 3]
```diff
@@ -192,12 +193,27 @@ func (c *Client) targetToTimeseries(ctx context.Context, target string, from str
...
+			// If not last point and interpolation enable, linear interpolate intermediate points.
+			if (j+1) < len(renderResponse.Datapoints) && c.cfg.Read.MaxPointDelta != time.Duration(0) {
+				intervalSecond := int64(c.cfg.Read.MaxPointDelta.Seconds())
+				nextDatapoint := renderResponse.Datapoints[j+1]
+				if nextDatapoint.Value == nil {
+					continue
+				}
+
+				variation := (*nextDatapoint.Value - *datapoint.Value) / float64(nextDatapoint.Timestamp-datapoint.Timestamp)
+				for timestamp := datapoint.Timestamp + intervalSecond; timestamp < nextDatapoint.Timestamp; timestamp += intervalSecond {
```
Comment (iksaif) [inline client/graphite/read.go]: Maybe something like that would be more clear ?

```
for i := 0; i < delta / interval; i++ {
timestamp := i * interval
value := *datapoint.Value + float64(timestamp-datapoint.Timestamp)*variation
}
```

File: client/graphite/read.go (hunk 2 above)
Comment (Thib17) [reply client/graphite/read.go]: Done.

File: client/graphite/config/cli.go (hunk 1 above)
Comment (Thib17) [reply client/graphite/config/cli.go]: Done.

File: client/graphite/read.go (hunk 3 above)
Comment (Thib17) [reply client/graphite/read.go]: Done.

**Pull request merged**